packages/core/test/rehydration.test.ts

```diff
@@ -122,6 +122,37 @@ describe('rehydration', () => {
 
       expect(actual).toEqual([]);
     });
+
+    it('should handle incompatible state value (shallow)', () => {
```

Comment:
those tests should be moved to another location - we don't even attempt to rehydrate here since this is an early error so those tests don't quite belong to `rehydration.test.ts`

I would also suggest to improve those test titles. "Handle" implies to me that we somehow manage to deal with such a situation but in fact we just throw - so the simplest adjustment would be to rephrase this to smth like:
```suggestion
    it('should error on incompatible state value (shallow)', () => {
```